front-page.php

```diff
@@ -114,6 +114,7 @@
           </label>
         </div>
       </div>
+      <input type="hidden" name="referer" value="<?php echo $_SERVER['HTTP_REFERER']; ?>" />
```

Comment:
Do you know if the browser will carry over the `HTTP_REFERRER` through the `XMLHttpRequest`? I'm wondering if we can hide this away by checking this server side in the request handler